unit_testing/test_labels.py

```diff
@@ -0,0 +1,208 @@
+    expected.data[0, 0, 0] = 7
+    expected.data[0, 1, 2] = 5
+
+    b = sct_labels.create_labels_empty(a, labels)
+
+    diff = b.data == expected.data
+    assert diff.all()
+
+
+@pytest.mark.parametrize("test_image", test_images)
+def test_create_labels(test_image):
+    a = test_image.copy()
+    labels = [Coordinate(l) for l in [[0, 1, 0, 99], [0, 1, 2, 5]]]
+
+    b = sct_labels.create_labels(a, labels)
+
+    assert b.data[0, 1, 0] == 99
+    assert b.data[0, 1, 2] == 5
+
+
+@pytest.mark.parametrize("test_seg", [seg_img])
+def test_create_labels_along_segmentation(test_seg):
+    a = test_seg.copy()
+    labels = [(5, 1), (14, 2), (23, 3)]
+
+    b = sct_labels.create_labels_along_segmentation(a, labels)
+
+    assert b.orientation == a.orientation
+    # TODO [AJ] implement test
+
+
+@pytest.mark.parametrize("test_image", test_images)
+def test_cubic_to_point(test_image):
+    a = test_image.copy()
+    sct_labels.cubic_to_point(a)
+    # TODO [AJ] implement test
+
+
+@pytest.mark.parametrize("test_image", test_images)
+def test_increment_z_inverse(test_image):
+    a = test_image.copy()
+    sct_labels.increment_z_inverse(a)
+    # TODO [AJ] implement test
+
+
+@pytest.mark.parametrize("test_seg,test_labels", [(seg_img, labels_img)])
+def test_labelize_from_disks(test_seg, test_labels):
+    seg = test_seg.copy()
+    ref = test_labels.copy()
+
+    sct_labels.labelize_from_disks(seg, ref)
+    # TODO [AJ] implement test
+
+
+@pytest.mark.parametrize("test_image", test_images)
+def test_label_vertebrae(test_image):
+    a = test_image.copy()
+    sct_labels.label_vertebrae(a)
+    sct_labels.label_vertebrae(a, [1, 2, 3, 4])
+    # TODO [AJ] implement test
+
+
+@pytest.mark.parametrize("test_image", test_images)
+def test_compute_mean_squared_error(test_image):
+    src = test_image.copy()
+    ref = test_image.copy()
+
+    for z in range(3):
+        for y in range(2):
+            for x in range(1):
+                ref.data[x, y, z] *= 10
+
+    sct_labels.compute_mean_squared_error(src, ref)
+    # TODO [AJ] implement test
+
+
+@pytest.mark.skip(reason="Too long to run on large image!")
+@pytest.mark.parametrize("test_image", test_images)
+def test_remove_missing_labels(test_image):
+    src = test_image.copy()
+    ref = test_image.copy()
+
+    expected = test_image.copy()
+
+    # introduce 2 discrepancies
+    src.data[0, 0, 0] = 99
+    src.data[0, 1, 2] = 99
+
+    ref.data[0, 0, 0] = 1
+    ref.data[0, 1, 2] = 1
+
+    # manually set expected
+    expected.data[0, 0, 0] = 0
+    expected.data[0, 1, 2] = 0
+
+    res = sct_labels.remove_missing_labels(src, ref)
+    diff = res.data == expected.data
+
+    assert diff.all()
+
+
+@pytest.mark.parametrize("test_image", test_images)
+def test_continuous_vertebral_levels(test_image):
+    a = test_image.copy()
+    b = sct_labels.continuous_vertebral_levels(a)
+
+    # check that orientation is maintained
+    assert b.orientation == a.orientation
+    # TODO [AJ] implement test
```

Comment:
i would run it and hard-code the output (eg: test a portion of the image that corresponds to the continuously-labeled segmentation, and assert if the vectors are the same)